src/router/CrossChainRouter.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.20;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+// upgrading contracts
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+import {IUltraLRT} from "../interfaces/IUltraLRT.sol";
+
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {IXERC20} from "src/xERC20/interfaces/IXERC20.sol";
+import {IXERC20Lockbox} from "src/xERC20/interfaces/IXERC20Lockbox.sol";
+import {XUltraLRT} from "src/xERC20/contracts/XUltraLRT.sol";
+
+contract CrossChainRouter is Initializable, UUPSUpgradeable, PausableUpgradeable, OwnableUpgradeable {
+    using SafeTransferLib for ERC20;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(address _governance) public initializer {
+        __Pausable_init();
+        __Ownable_init(_governance);
+    }
+
+    function _authorizeUpgrade(address newImplementation) internal override onlyOwner {}
+
+    function pause() public onlyOwner {
+        _pause();
+    }
+
+    function unpause() public onlyOwner {
+        _unpause();
+    }
+
+    function transferRemoteUltraLRT(address _ultraLRT, address _lockbox, uint32 _destination, uint256 _amount)
```

Comment:
can you not put lockbox as a parameter? we need to ensure that we don't get hacked at the SDK level. I don' think our lockbox will change easily--just set this as an upgradable param in the router

Reply:
dropped.